src/controlplane/client/connector.go

```diff
@@ -0,0 +1,274 @@
+
+	case strings.EqualFold(endpoint.Auth.Type, mTLSAuth):
+		tlsConfig, err := dp.getTLSConfigFromSecret(endpoint.Auth.MTLS)
+		if err != nil {
+			return fmt.Errorf("could not load TLS configuration: %w", err)
+		}
+
+		dp.logger.Debugf("MTLS auth is used for endpoint %q", endpoint.URL)
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+
+	default:
+		return fmt.Errorf("authorization type not supported: %q", endpoint.Auth.Type)
+	}
+
+	return nil
+}
+
+func (dp *defaultConnector) getTLSConfigFromSecret(mTLSConfig *config.MTLS) (*tls.Config, error) {
+	if mTLSConfig == nil {
+		return nil, fmt.Errorf("mTLS config cannot be nil")
+	}
+
+	if mTLSConfig.TLSSecretName == "" {
+		return nil, fmt.Errorf("mTLS secret name cannot be empty")
+	}
+
+	namespace := mTLSConfig.TLSSecretNamespace
+	secretName := mTLSConfig.TLSSecretName
+	if namespace == "" {
+		dp.logger.Debugf("TLS Secret name configured, but not TLS Secret namespace. Defaulting to `default` namespace.")
+		namespace = "default"
+	}
+
+	secret, err := dp.kc.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", secretName, err)
+	}
+
+	var cert, key, cacert []byte
+
+	var ok bool
+	if cert, ok = secret.Data["cert"]; !ok {
+		return nil, fmt.Errorf("could not find TLS certificate in `cert` field in secret %q", secretName)
+	}
+
+	if key, ok = secret.Data["key"]; !ok {
+		return nil, fmt.Errorf("could not find TLS key in `key` field in secret %q", secretName)
+	}
+
+	cacert, hasCACert := secret.Data["cacert"]
+	insecureSkipVerifyRaw, hasInsecureSkipVerify := secret.Data["insecureSkipVerify"]
+
+	if !hasCACert && !hasInsecureSkipVerify {
+		return nil, fmt.Errorf("both cacert and insecureSkipVerify are not set. One of them need to be set to be able to call ETCD metrics")
+	}
+
+	// insecureSkipVerify is set to false by default, and can be overridden with the insecureSkipVerify field
+	insecureSkipVerify := false
+	if hasInsecureSkipVerify {
+		insecureSkipVerify = strings.ToLower(string(insecureSkipVerifyRaw)) == "true"
+	}
+
+	tlsConfig, err := parseTLSConfig(cert, key, cacert, insecureSkipVerify)
+	if err != nil {
+		return nil, fmt.Errorf("parsing TLS config: %w", err)
+	}
+
+	return tlsConfig, nil
+}
+
+func parseTLSConfig(certPEMBlock, keyPEMBlock, cacertPEMBlock []byte, insecureSkipVerify bool) (*tls.Config, error) {
+	cert, err := tls.X509KeyPair(certPEMBlock, keyPEMBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	clientCertPool := x509.NewCertPool()
+
+	if len(cacertPEMBlock) > 0 {
+		clientCertPool.AppendCertsFromPEM(cacertPEMBlock)
+	}
+
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{cert},
+		RootCAs:            clientCertPool,
+		InsecureSkipVerify: insecureSkipVerify,
+	}
+
+	tlsConfig.BuildNameToCertificate()
+
+	return tlsConfig, nil
+}
+
+type connParams struct {
```

Comment:
in the Kubelet client we take connParams and save into client.url and client.doer the data coming from that. 
I am fine with both approaches, but we need to follow the same